velox/vector/BaseVector.cpp

```diff
@@ -558,6 +559,7 @@ void BaseVector::ensureWritable(
     } else {
       result = BaseVector::create(type, rows.size(), pool);
     }
+    result->clearDataDependentFlags();
```

Comment:
Curious why we would prefer to reset flags in VectorPool instead of its caller here? BaseVector::create() actually set the data-dependent flags for some data types too (See BaseVector::createInternal()), so if we don't clear the flags in the caller here, we need to change both BaseVector::create() and VectorPool::get().

Reply:
According to offline discussion, VectorPool may be used by itself, so we should make sure data-dependent flags are cleared by it. On the other hand, we're OK with BaseVector::create() setting some data-dependent flags since those are passed to vector constructors and should be legit. Therefore, I removed result->clearDataDependentFlags() here.